### PR TITLE
[IMP] http_routing, *: move error pages to http_routing

### PR DESCRIPTION
**Pull request by fja-odoo**
* = portal, website, test_website, account, sms

Portal an Survey error pages were ugly default pages.
Now the error pages will be the same as the website ones.

Also fixing single module builds tests

Some url with /my or /hr for example are thaught to be a language code.
Now only true and available language code will be used.
Also we will check if the language code is installed.

task-2059969

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fja-odoo** [comment]
@seb-odoo

**Comment by fja-odoo** [comment]
@JKE-be Can you take a look at this , it will move all website error pages and exception handling to portal. That way we have all the "beautiful pages" on portal too. Also it will fix some tests on the solo module builds. Those tests are expecting 400 error on portal but since website is not installed it fails.
Here are some of the tests failing: http://runbot.odoo.com/runbot/build/599811

File: addons/portal/models/ir_http.py [hunk 1]
```diff
@@ -12,3 +23,112 @@ class IrHttp(models.AbstractModel):
...
+        if isinstance(exception, exceptions.except_orm):
+            values['error_message'] = exception.name
+            code = 400
+            if isinstance(exception, exceptions.AccessError):
+                code = 403
+
+        elif isinstance(exception, QWebException):
+            values.update(qweb_exception=exception)
+
+            # retro compatibility to remove in 12.2
```
Comment (JKE-be) [inline addons/portal/models/ir_http.py]: time to remove

File: addons/portal/models/ir_http.py [hunk 2]
```diff
@@ -12,3 +23,112 @@ class IrHttp(models.AbstractModel):
...
+
+        values.update(
+            status_message=werkzeug.http.HTTP_STATUS_CODES[code],
+            status_code=code,
+        )
+
+        return (code, values)
+
+    @classmethod
+    def _get_values_500_error(cls, env, values, exception):
```
Comment (JKE-be) [inline addons/portal/models/ir_http.py]: keep reset mechanism in website only

File: addons/http_routing/views/http_routing_template.xml [hunk 3]
```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="http_error">
+        <t t-call="web.frontend_layout">
+            <div id="wrap">
+                <div class="oe_structure">
+                    <h1 class="container mt32"><t t-esc="status_code"/>: <t t-esc="status_message"/></h1>
+                </div>
+                <t t-if="editable or debug">
```
Comment (rdeodoo) [inline addons/http_routing/views/http_routing_template.xml]: somehow `debug` seems to be always true (was already the case before your commit.. checked in 12.4 and same bug) while `request.session.debug` has the correct value.. (it should be the same as debug is an alias to request.session.debug added in prepare_qcontext)
So the traceback is always shown and not only in debug
Maybe we can change it here so 12.5 is clean and I'll investigate in 12.4 what's going on

**Comment by JKE-be** [comment]
@robodoo r+

**Comment by Xavier-Do** [comment]
robodoo r- looks like it's conflicting with #36555 where a logger.debug was added, either remove it or keep the logger

**Comment by JKE-be** [comment]
@robodoo retry r+

**Comment by Xavier-Do** [comment]
robodoo r+

**Comment by Xavier-Do** [comment]
@robodoo ok

**Pull request closed without merging**